packages/mesos-client/src/createEventStreamConnection.js

```diff
@@ -0,0 +1,34 @@
+import Rx from "rxjs";
+import { read } from "recordio";
+import { XHRConnection } from "connection-manager";
+
+import observableFromXhrConnection from "./observableFromXhrConnection";
+
+function extractRecords({ buffer = "", position = 0 } = {}, data = "") {
+  // XHRConnection appends to its responseText on every progress event
+  // but we need the data chunk by chunk to emit records as they come
+  // thus we track the position in data that we've already processed
+  const chunk = data.substr(position);
+  const [records, rest] = read(buffer + chunk);
+
+  return { records, buffer: rest, position: position + chunk.length };
+}
+
+export default function createEventStreamConnection(baseUrl) {
+  const connection = new XHRConnection(
```

Comment:
Not relevant to your PR, but now that I'm seeing the multiple arguments approach to configuring an `XHRConnection` it doesn't feel right. We're used to a single options Object for configuring `XMLHttpRequest` (and most other request libraries) and it feels bizarre to do it this way. cc/ @orlandohohmeier @juliangieseke

Reply:
Adjusted in #2361